tests/NuGet.IndexingTests/PackageAnalyzerTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Indexing;
+using NuGet.IndexingTests.TestSupport;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class PackageAnalyzerTests
+    {
+        [Theory, MemberData("TheoryData")]
+        public void AddsCorrectFieldAnalyzers(string field, string text, TokenAttributes[] expected)
```

Comment:
At least in VS, the failures and the successes are reported separately (including the assertion failure, the test output, etc).